src/utils/polygon.h

```diff
@@ -991,9 +992,40 @@ class Polygons
         }
     }
 
-    void fixSelfIntersections()
+    void resize(const coord_t& numerator, const coord_t divisor)
     {
-        ClipperLib::SimplifyPolygons(paths);
+        if (numerator == 1 && divisor == 1)
+        {
+            return;
+        }
+
+        Polygons& thiss = *this;
+        for (size_t p = 0; p < size(); p++)
+        {
+            for (Point& pt : thiss[p])
+            {
+                pt.X = (pt.X * numerator) / divisor;
+                pt.Y = (pt.Y * numerator) / divisor;
```

Comment:
The `Point` class also supports the operator `*` which does the same.

Reply:
I'll have to look into this, since I'm not the original author. I think @rburema wrote it, maybe he can comment on this.

Reply:
Changed